internal/scylla/client.go

```diff
@@ -237,24 +194,37 @@ func (c *Client) callAPI(ctx context.Context, method, path string, reqBody, resT
 	}
 
 	if data.Error != "" {
-		err = makeError(data.Error, c.ErrCodes, resp)
-		tflog.Trace(ctx, "api returned error: "+err.Error(), map[string]interface{}{
-			"code":   resp.StatusCode,
-			"status": resp.Status,
-			"body":   buf.String(),
-			"error":  err.Error(),
-		})
-		return err
+		return makeAPIError(data.Error, c.ErrCodes, resp.Request.URL.String(), method, resp.StatusCode, retryAfter)
 	}
 
-	tflog.Trace(ctx, "api call succeeded", map[string]interface{}{
-		"code":   resp.StatusCode,
-		"status": resp.Status,
-		"body":   buf.String(),
-	})
 	return nil
 }
 
+func (c *Client) callAPI(ctx context.Context, method, path string, reqBody, resType interface{}, query ...string) (err error) {
+	defaultWaitTime := 100 * time.Millisecond
+	for retry := 0; retry <= retriesAllowed; retry++ {
+		err = c.callAPIonce(ctx, method, path, reqBody, resType, query...)
+		if err == nil {
+			return nil
+		}
+		isRetryable, timeToSleep := getErrRetryInfo(err)
+		if !isRetryable {
+			return err
+		}
+		if timeToSleep == 0 {
+			defaultWaitTime *= 2
+			if defaultWaitTime > 5*time.Second {
+				defaultWaitTime = 5 * time.Second
+			}
```

Comment:
any reason to hardcode this backoff strategy and why were this exact numbers chosen?

Reply:
there should be some backoff, out of the case you can see that exponential is the best for it, rest is just eyeballed.